src/main/java/core/basesyntax/transactions/OperationStrategy.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax.transactions;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import java.util.Map;
+
+public class OperationStrategy {
+    private static final String NULL_VALUE = "The value cannot be null";
+    private static final String UNDER_ZERO_QUANTITY = "The quantity must be positive";
+    private final Map<Operation, OperationHandler> handlers;
+
+    public OperationStrategy(Map<Operation, OperationHandler> handlers) {
+        this.handlers = handlers;
+    }
+
+    public OperationHandler handler(FruitTransaction fruitTransaction) {
+        checkOnNull(fruitTransaction);
+        OperationHandler operationHandler = handlers.get(fruitTransaction.getOperation());
+        operationHandler.processTransaction(fruitTransaction);
+        return operationHandler;
+    }
+
+    private void checkOnNull(FruitTransaction fruitTransaction) {
+        if (fruitTransaction == null
+                || fruitTransaction.getFruit() == null
+                || fruitTransaction.getOperation() == null) {
+            throw new NullPointerException(NULL_VALUE + fruitTransaction);

```

Comment:
I would recommend you never throw `NullPointerException` by yourself.

The main reason not to do this, as Thorbjørn Ravn Andersen says in a comment below, is that you don't wan't to mix 'real, bad NPEs' with NPEs thrown intentionally.

So, until you're confident that you're able to recognize 'valid' NPE, I'd recommend to use `IllegalArgumentException` when you want to tell to your API user that null is not a valid argument value. Your method's behavior when illegal null-parameter passed should be documented.
Check [this](https://stackoverflow.com/questions/3322638/is-it-okay-to-throw-nullpointerexception-programmatically) article please.